### ISSUE TITLE
Use case: Tiling visualization

### ISSUE THREAD

**Issue by VisenDev** [foxnne/pixi#46]
It would be useful to be able to have a way to visualize how a sprite would look tiled. This would be very useful for things like creating tilesets

**Comment by foxnne** [comment]
Agreed! I haven't gotten to the point in my game project to need this quite yet so I haven't looked into how I want this to look yet, but I'm very open to ideas. I'm considering using Tiled or similar and need to examine the workflow and how to bridge between the two. PyxelEdit, which I used to use, had this built in and we could do similar for tiled sprite editing.

**Comment by slimsag** [comment]
For a pixi file like this in Lord of Zero:

<img width="909" alt="image" src="https://github.com/foxnne/pixi/assets/3173176/b9b427cf-3f20-4325-a76b-4e2a0f36463e">

I think what I want is really the ability to say "hey, this tile over <here>? it's exactly the same as this tile over <here> - when I make edits to it, make edits to all the others. Consider it a clone."

Also, the packed image ends up with a fair amount of wasted space because I have the tile duplicated so I can see it visually:

<img width="1392" alt="image" src="https://github.com/foxnne/pixi/assets/3173176/0bb8cf3e-a021-49a1-bbcb-0d0086bc256c">

It would be nice if these tiles "considered a clone" would be skipped during export too.

**Comment by foxnne** [comment]
That sounds good! I had a similar idea myself. The other option that might be just as nice is to have a setting for the "flip book" panel at the bottom, to instead of drawing individual scrolling cards, draw the selected tile 9 times, a 3x3 grid, all duplicating the selected sprite. Then also make each tile editable directly. Then you could select a tile to begin drawing, and draw on any of the 3x3 grid to draw the tiling sprite. You'd then only need the single tile in the main drawing so it wouldn't be duplicated.

**Comment by slimsag** [comment]
That seems like it would be a good approach, too. The limitation I see there is that if you are working on this:

<img width="305" alt="image" src="https://github.com/foxnne/pixi/assets/3173176/99841c0c-b60d-4b86-9448-b84eff7d59be">

Or this:

<img width="374" alt="image" src="https://github.com/foxnne/pixi/assets/3173176/f51d411d-c870-4001-8957-33e6d69e1d82">

In a 3x3 view, it would be hard to see the 'overall picture' there, since not all of the 3x3 are the same, so you'd probably need to duplicate anyway to see that - or view them in a level editor.

Similarly, being able to see transitions from one tile to another - like this telling me the rick tile and ground tile don't have the same height:

<img width="311" alt="image" src="https://github.com/foxnne/pixi/assets/3173176/b81ef907-6e41-4837-9b24-e85338bec3b8">

I think the copy+paste cloning would be more powerful for those situations, but would have the drawbacks you mentioned (it uses a slot in your actual grid-although those are 'cheap', and is maybe a bit more complex to setup just for the 3x3 case which is very common)

**Comment by foxnne** [comment]
@VisenDev @slimsag [This commit](https://github.com/foxnne/pixi/commit/4cf073279707a8737cd2f1985108dd261c7827f7) added a setting under the `View` menu option to change the flip book view from sequential to grid, which allows easily making seamless tiles. All the usual controls work with this new view.

As @slimsag mentioned, there are some cases where it would probably be really handy to be able to make tiles a reference to another tile, I need to think on this a bit more as I don't want things to become hard to follow. Tiles that are references need some visualization.

Anyway, hopefully this helps!

**Comment by foxnne** [comment]
Since rewrite this no longer exists. Pixi will need to reimplement the `flipbook` pane and also allow tiling sprite creation and editing. It would be really great to also support "referencing" a tile, which would mean any edits to any location of that tile would propagate, and only one instance of the tile would be packed.
